src/fft.rg

```diff
@@ -172,105 +241,279 @@ function fft.generate_fft_interface(itype, dtype)
-        var output_base = get_base(rect_t(output.ispace.bounds), __physical(output)[0], __fields(output)[0])
-        var lo = input.ispace.bounds.lo:to_point()
-        var hi = input.ispace.bounds.hi:to_point()
-        var n : int[dim]
-        ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)]
-        var cufft_p : cufft_c.cufftHandle
-        var ok = cufft_c.cufftPlanMany(
-          &cufft_p,
-          dim,
-          &n[0],
-          [&int](0), 0, 0, -- inembed, istride, idist
-          [&int](0), 0, 0, -- onembed, ostride, odist
-          cufft_c.CUFFT_C2C,
-          1 -- batch
-        )
-        regentlib.assert(ok == cufft_c.CUFFT_SUCCESS, "cufftPlanMany failed")
-        return cufft_p
-      else
+      if c.legion_processor_kind(proc) ~= c.TOC_PROC then
         regentlib.assert(false, "make_plan_gpu must be executed on a GPU processor")
       end
+
+      var i = c.legion_processor_address_space(proc)
+      regentlib.assert(address_space == i, "make_plan_gpu must be executed on a processor in the same address space")
+
+      -- Get input and output bases
+      var input_base = get_base_in(rect_in_t(input.ispace.bounds), __physical(input)[0], __fields(input)[0]).base
+      var output_base = get_base_out(rect_out_t(output.ispace.bounds), __physical(output)[0], __fields(output)[0]).base
+      var lo = input.ispace.bounds.lo:to_point()
+      var hi = input.ispace.bounds.hi:to_point()
+      var n : int[dim] -- n is an array of size dim with the size of each dimension in the entries
+      ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)]
+
+      -- Create plans
+      if float_to_complex32_transform then
+        cufft_assert(cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_R2C, 1))
+      elseif complex32_to_complex32_transform then
+        cufft_assert(cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_C2C, 1))
+      elseif double_to_complex64_transform then
+        cufft_assert(cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_D2Z, 1))
+      elseif complex64_to_complex64_transform then
+        cufft_assert(cufft_c.cufftPlanMany(&p.cufft_p, dim, &n[0], [&int](0), 0, 0, [&int](0), 0, 0, cufft_c.CUFFT_Z2Z, 1))
+      end
     end
   end
 
-  -- Important: overwrites input/output!
-  __demand(__inline)
-  task iface.make_plan(input : region(ispace(itype), dtype),
-                       output : region(ispace(itype), dtype),
+  --- Creates the FFT plan. This is the __inline version of the task should the user wish to use that. Entry point into GPU functionality as well.
+  -- @param input Input region.
+  -- @param output Output region.
+  -- @param plan Plan region.
+  -- @note Calls `make_plan_gpu` if necessary.
```

Comment:
```suggestion
```

`make_plan_gpu` is not part of the public interface and we should never mention it in the documentation or API comments.